### PR TITLE
Refactor: gateway updates

### PR DESCRIPTION
**Pull request by jonwaldstein**
Depends on https://github.com/impress-org/givewp/pull/6658

Related: https://github.com/impress-org/givewp/pull/6695, https://github.com/impress-org/givewp/pull/6699

## Description

Updates to the example gateway:
- Change Acme to Example so there's no confusion with Acme Payments
- Remove subscription logic for now
- Update offsite route method name & logic
- Add filters for `$gatewayData` param
- Add stepped comments to make things more clear
- Simplify the example code

## Affects

Everything

## Visuals

N/A

## Testing Instructions

- Install plugin / pull changes
- Activate plugin
- Enable offsite and onsite gateways
- Donate using a donation form with both gateways for one-time and recurring
- Make sure everything works as expected.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: class-onsite-example-gateway.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            // Step 1: Validate any data passed from the gateway fields in $gatewayData.  Throw the PaymentGatewayException if the data is invalid.
+            if (empty($gatewayData['example_payment_id'])) {
+                throw new PaymentGatewayException('Example payment ID is required.');
+            }
+
+            // Step 2: Create a payment with your gateway.
+            $response = $this->exampleRequest(['transaction_id' => $gatewayData['example_payment_id']]);
+
+            // Step 3: Return a command to complete the donation.
+            return new PaymentComplete($response['transaction_id']);
```
Comment (JasonTheAdams) [inline class-onsite-example-gateway.php]: Should we leave a note here that `PaymentProcessing` can also be used if the payment needs to be marked as complete later via webhooks?

File: givewp-example-gateway.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        // Step 1: Validate the data you need to add to the gatewayData (This comes from your gateway fields).
+        if (isset($_POST['example-gateway-id'])) {
+            // Step 2: Assign the sanitized data to the gatewayData as an array item.
+            $gatewayData['example_payment_id'] = sanitize_text_field($_POST['example-gateway-id']);
+        }
+
+        // Step 3: Return the gatewayData array.
+        return $gatewayData;
+    }
+);
```
Comment (JasonTheAdams) [inline givewp-example-gateway.php]: @jonwaldstein I was hoping we would make this automatic, not requiring this extra hook. The hook was only added for compatibility purposes. I believe if they include a `$_REQUEST['gatewayData']` value it automatically grabs it, right?

**Review by JasonTheAdams — COMMENTED** [review]
Left a couple of questions, but this is a good start for getting the Gateway API documented. Thanks for all your work @Benunc and @jonwaldstein!

File: givewp-example-gateway.php (hunk 2 above)
Comment (jonwaldstein) [reply givewp-example-gateway.php]: That's right, they should actually be able to update their input name to be inside gatewayData and they can retrieve it without using the filter.

```
<div><input type='text' name='gatewayData[example-gateway-id]' placeholder='Example gateway field' /></div>
```

Comment (jonwaldstein) [reply givewp-example-gateway.php]: That worked [b1bc63a](https://github.com/impress-org/givewp-example-gateway/pull/1/commits/b1bc63ab2729f95575a49f15255e3beffc3adebb)

File: class-onsite-example-gateway.php (hunk 1 above)
Comment (Benunc) [reply class-onsite-example-gateway.php]: yes. let's add that, @JasonTheAdams is that something I can do?

Comment (JasonTheAdams) [reply class-onsite-example-gateway.php]: @Benunc Sure! Just add a comment with a line identical to this one that uses `PaymentProcessing` instead. Simple as that! And, of course, explain _why_ one would use that command. 😁

**Review by JasonTheAdams — APPROVED** [review]
Looks great! Good work, y'all!

**Pull request merged**